pkg/collector/listeners/ecs.go

```diff
@@ -0,0 +1,232 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build docker
+
+package listeners
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+	"github.com/DataDog/datadog-agent/pkg/util/ecs"
+	log "github.com/cihub/seelog"
+)
+
+// ECSListener implements the ServiceListener interface for fargate-backed ECS cluster.
+// It pulls its tasks container list periodically and checks for
+// new containers to monitor, and old containers to stop monitoring
+type ECSListener struct {
+	task       ecs.TaskMetadata
+	services   map[string]Service // maps container IDs to services
+	newService chan<- Service
+	delService chan<- Service
+	stop       chan bool
+	m          sync.RWMutex
```

Comment:
you're not using the mutex when modifying the `services` map

Reply:
should be good now